### PR TITLE
Fix #97 - update azure pipeline to actually build both Windows and Mac

### LINKED ISSUE #97: Automate builds with a cloud service
I'm investigating automating builds using a cloud service -- Azure Pipelines apparently allows for both Windows and macOS for free for open source projects; I almost created a working pipeline for macOS, it would be nice to get 'nightly' builds at least for now, and have some way of automating release. Unless someone else has a better idea.

I started looking around after I noticed Mac Stadium allowed free macOS compute time for open source projects: https://www.macstadium.com/opensource

If anyone knows any other services offering free compute time for Windows + Visual Studio build tools and/or macOS + XCode build tools, please do post them here!

### PR DESCRIPTION
**Pull request by kzantow**

### CODE REVIEW

**Comment by esaruoho** [comment]
@kzantow weird, looks like the checks failed.. what's up with that?

**Comment by kzantow** [comment]
@esaruoho no, it's expected. There is a file that needs to be provided by the azure pipeline secure file mechanism, which is necessary to download the VST 2 sources. I haven't figured out a way to make that optional. But as I was noting to @kurasu it probably should NOT be running this pipeline at all! Otherwise any secret that is used for a release or such could be stolen by any random person submitting a PR... in a scenario like this, where anyone can submit PRs to a public repo, the ideal situation is that it always runs a specific pipeline for them, e.g. `master/azure-pipeline.yaml` and _not_ the PR version. I'm looking for a way to configure things like that, but I'm not sure it's supported... in which case either disabling PR builds or coming up with an alternate way to do releases will be necessary. :-/

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,19 +1,144 @@
...
 # https://aka.ms/yaml
 
 trigger:
 - master
+- releases/*
 
-pool:
-  vmImage: 'Ubuntu-16.04'
+pr:
+- master
```
Comment (kzantow) [inline azure-pipelines.yml]: might need to make this `none` :-/

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,19 +1,144 @@
...
+    vmImage: $(imageName)
+
+  steps:
+  - checkout: self
+    fetchDepth: 1
+    # submodules: recursive # can't do submodules here b'cuz depth=1 fails with Github
+
+  - task: DownloadSecureFile@1
+    inputs:
+      secureFile: get-vst2.sh
```
Comment (kzantow) [inline azure-pipelines.yml]: doesn't seem to be a way to make this optional :-/

**Comment by baconpaul** [comment]
Does this need to be merged now?

**Comment by kzantow** [comment]
@baconpaul yeah, this is mostly ready to merge. just need to coordinate with @kurasu to add a script to the Azure pipeline

**Comment by kzantow** [comment]
@kurasu @baconpaul @esaruoho FYI this looks to be working fine for now, and no problem to build on PRs now. It does *not* build VST 2, however, but does build installer artifacts without VST 2 that you can download by following the links to the Azure Pipeline run. As noted before, I created a separate repository for releases. That can be changed to do nightly builds or other things as needed, but not part of this PR.

**Comment by baconpaul** [comment]
Great! So if I merge this will PRs just all get built automatically? Any downside to me doing so? It all looks good to me just don't know if you need a different step.

Ooh also those nugget chocolatey lines for innosetup are useful for my powershell equivalent of build-os I'm increasingly using not he command line and may finish up one day.

**Comment by baconpaul** [comment]
@esaruoho I do not believe there is a good answer to that question available to us today. But I would love a build on every pr even if it skips vst2 so would merge this if that’s what it does!

**Comment by kzantow** [comment]
@esaruoho this pipeline is more of a 'validate the PR' pipeline, makes sure it builds and creates installers which could be downloaded and viewed (although I checked and there may be some permission issue to fix in Azure).

**Comment by baconpaul** [comment]
Ok merged!

**Pull request merged**